snap/backend/src/utils/mosaicUtils.js

```diff
@@ -0,0 +1,65 @@
+import symbolClient from '../services/symbolClient.js';
+import stateManager from '../stateManager.js';
+
+const mosaicUtils = {
+	/**
+     * Update mosaic info in state if given mosaic ids not exist in mosaicInfo.
+     * @param {object} state - The state object.
+     * @param {Array<string>} mosaicIds - The mosaic ids.
+     * @returns {Promise<void>} - A promise that resolves when mosaic info is updated.
+     */
+	updateMosaicInfo: async (state, mosaicIds) => {
+		try {
+			const { network, mosaicInfo } = state;
+
+			// filter out mosaic id didn't exist in mosaicInfo
+			const missingMosaicIds = mosaicIds.filter(mosaicId => !mosaicInfo[mosaicId]);
+
+			if (0 === missingMosaicIds.length)
+				return;
+
+			const client = symbolClient.create(network.url);
+
+			const mosaics = await client.fetchMosaicsInfo(missingMosaicIds);
+
+			// update networkName in mosaic info
+			Object.keys(mosaics).forEach(mosaicId => {
+				mosaics[mosaicId].networkName = network.networkName;
+			});
```

Comment:
Why is the mosaic need to have a network name as state?

Reply:
`mosaicInfo` is the cache in the state (global), which will be shared between the network (testnet/mainnet).

The reason added `networkName` is for filtering purposes, which switch between networks.

Reply:
For security reasons, I am unsure if mainnet and testnet would share the same backend.

Reply:
I think we can decide later after I make a release for the internal team test.